src/neo/SmartContract/NotifyEventArgs.cs

```diff
@@ -29,7 +29,12 @@ public void FromStackItem(StackItem stackItem)
 
         public StackItem ToStackItem(ReferenceCounter referenceCounter)
         {
-            return new Array(referenceCounter) { ScriptHash.ToArray(), EventName, State };
+            return new Array(referenceCounter)
+            {
+                ScriptHash.ToArray(),
+                EventName,
+                State.DeepCopy()
```

Comment:
Maybe only with this it's enough? We make two deep copy per notifications.

Reply:
We have to make two deep copy. Otherwise the sender can modify it after sent.